src/nb_enum.cpp

```diff
@@ -107,6 +117,48 @@ static PyObject *nb_enum_init(PyObject *, PyObject *, PyObject *) {
     return 0;
 }
 
+static bool nb_enum_set_value(PyObject *self, PyObject *value) {
```

Comment:
It's a nice idea, but the amount of library code needed to realize the feature seems high to me (esp. the `nb_enum_set_value` function).

Could this feature also be realized on the user's end using a custom `nb::init<>` constructor? (With some library to perhaps automate the creation of that)